### ISSUE TITLE
AB OpenGL errors, no display

### ISSUE THREAD

**Issue by letheed** [kmkolasinski/AwesomeBump#130]
AwesomeBump fails to display any image.

![awesomebump](https://user-images.githubusercontent.com/5330692/30176915-5dfb23fc-9404-11e7-8c13-51a47575a18e.png)

Here are the logs, lots of OpenGL errors.

[AwesomeBump.txt](https://github.com/kmkolasinski/AwesomeBump/files/1285596/AwesomeBump.txt)
[AwesomeBumpGL330.txt](https://github.com/kmkolasinski/AwesomeBump/files/1285598/AwesomeBumpGL330.txt)

```bash
$ inxi -GS
System:    Host: liliane Kernel: 4.10.0-33-generic x86_64 (64 bit) Desktop: Gnome Distro: Ubuntu 17.04
Graphics:  Card: Advanced Micro Devices [AMD/ATI] Tahiti PRO [Radeon HD 7950/8950 OEM / R9 280]
           Display Server: X.Org 1.19.3 driver: FAILED: modesetting
           Resolution: 1920x1080@119.98hz, 1920x1080@50.00hz
           GLX Renderer: Gallium 0.4 on AMD TAHITI (DRM 2.49.0 / 4.10.0-33-generic, LLVM 4.0.0)
           GLX Version: 3.0 Mesa 17.1.8
```

**Comment by kmkolasinski** [comment]
Hi, you have Mesa driver, I cannot help you. AB just don't work with it.

**Comment by letheed** [comment]
Does it work with the amdgpu-pro driver?

**Comment by kmkolasinski** [comment]
Not sure, maybe. I haven't tried it before. You can try if it is not a problem for you.

**Comment by letheed** [comment]
Well it doesn't work either.

**Comment by kmkolasinski** [comment]
Just to be sure, do you have same errors?

**Comment by letheed** [comment]
Not exactly but the radeon driver supports OpenGL 4.5 whereas AMD's amdgpu-pro is at 4.1 at the moment. I've removed the time stamp so they diff properly.

[logs.zip](https://github.com/kmkolasinski/AwesomeBump/files/1286219/logs.zip)

**Comment by kmkolasinski** [comment]
Sorry, the openGL errors are not informative at all. You could try to build AB from source code. There is a script for ubuntu (you need Qt libraries > 5.0 version). Or you could try older version of AB. Maybe they will work.

**Comment by LinuxUserGD** [comment]
I get the same OpenGL 4.0 issue.
I've compiled the source code using Qt 5.10.0 on Linux. AB runs fine on Windows 10.

Output (compile):
[output.txt](https://github.com/kmkolasinski/AwesomeBump/files/1671552/output.txt)
Output (run):
[output-run.txt](https://github.com/kmkolasinski/AwesomeBump/files/1671553/output-run.txt)

```
System:    Host: user Kernel: 4.14.14-47.current x86_64 bits: 64
           Desktop: Budgie 10.4  Distro: Solus 3
Graphics:  Card: Intel HD Graphics 620
           Display Server: x11 (X.Org 1.19.5 )
           drivers: modesetting (unloaded: fbdev,vesa)
           Resolution: 1920x1080@60.05hz
           OpenGL: renderer: Mesa DRI Intel HD Graphics 620 (Kaby Lake GT2)
           version: 4.5 Mesa 17.3.1
```

**Comment by AO-StreetArt** [comment]
I'm running Centos7 with amdgpu-pro driver, Qt5 built from source.  I get the same error running latest source code checked out from the repo.

Interestingly, I am perfectly able to compile the v4 linux source code from the releases page and it runs without issue.

**Comment by kmkolasinski** [comment]
I think that it can be related somehow to your distribution. I have just run those commends:
```
git clone https://github.com/kmkolasinski/AwesomeBump.git
cd AwesomeBump
vim unixBuildScript.sh ## to set the MY_QT_PATH: Qt/5.8/gcc_64/bin/
./unixBuildScript.sh
cd ./Bin
```
And everything works fine, but I work on `Ubuntu 16.04.4 LTS` and I have also NVIDIA graphics card:

```
Checking OpenGL widget:
Widget OpenGL: "4.5"
Context valid: true
OpenGL information:
VENDOR: NVIDIA Corporation
RENDERER: GeForce GTX 950M/PCIe/SSE2
VERSION: 4.5.0 NVIDIA 384.111
GLSL VERSION: 4.50 NVIDIA
...
...
Reading new cube map: ("./Core/2D/skyboxes/1SaintLazarusChurch/posx.jpg", "./Core/2D/skyboxes/1SaintLazarusChurch/negx.jpg", "./Core/2D/skyboxes/1SaintLazarusChurch/posy.jpg", "./Core/2D/skyboxes/1SaintLazarusChurch/negy.jpg", "./Core/2D/skyboxes/1SaintLazarusChurch/posz.jpg", "./Core/2D/skyboxes/1SaintLazarusChurch/negz.jpg")
Generated number of mipmaps: 10
Skipping glWidget repainting during first Env. maps. load.
...
...
```
I don't remember the exact difference between V4 and V5 but maybe there was some small change which resulted with this issue. Are you able to compare the source codes between V4/V5? I guess it is related to OpenGL context. As far as I remember there were some changes with handling context between both versions.

**Comment by AO-StreetArt** [comment]
So I took a look at the difference between the versions, here are the things I thought might be relevant:

* gpuinfo.hpp and gpuinfo.cpp - Certain functions aren't implemented for AMD GPU's in v5, (GLint GpuInfo::getTotalMem()).  Still looking to see if that might create issues down the line.
* A bunch of classes under utils/contextinfo got added between versions

I'll try to keep looking sometime this week, no promises on being able to identify the issue though lol.

The tags I was comparing are ac92a815a8 and c11a1f029b
